File: examples/scala/src/test/scala/cuke4duke/examples/CalledSteps.scala
package cuke4duke.examples

import org.junit.Assert.assertTrue
import cuke4duke._

class CalledSteps extends ScalaDsl with EN {
  var magic = false

  Given("^it is (.*)$") { what: String =>
    if ("magic" == what)
      magic = true
  }

  Then("^magic should happen$") {
    assertTrue(magic);
  }
}

File: examples/guice/src/test/java/billing/CalledSteps.java
package billing;

import cuke4duke.annotation.I18n.EN.*;

import static org.junit.Assert.assertTrue;

public class CalledSteps {
    private boolean magic;

    @Given("^it is (.*)$")
    public void itIs(String what) {
        if(what.equals("magic")) {
            magic = true;
        }
    }

    @Then("^magic should happen$")
    public void magicShouldHappen() {
        assertTrue(magic);
    }
}
